projects/pswg/src/main/java/com/parzivail/pswg/item/ThermalDetonatorItem.java

```diff
@@ -0,0 +1,177 @@
+import net.minecraft.sound.SoundCategory;
+import net.minecraft.sound.SoundEvents;
+import net.minecraft.stat.Stats;
+import net.minecraft.text.Text;
+import net.minecraft.util.Hand;
+import net.minecraft.util.TypedActionResult;
+import net.minecraft.util.UseAction;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.Nullable;
+
+public class ThermalDetonatorItem extends Item implements ILeftClickConsumer
+{
+	public ThermalDetonatorItem(Settings settings)
+	{
+		super(settings);
+	}
+	public ThermalDetonatorEntity createThermalDetonator(World world, ItemStack stack, int life, boolean primed, PlayerEntity playerEntity) {
+		ThermalDetonatorEntity thermalDetonatorEntity = new ThermalDetonatorEntity(SwgEntities.Misc.ThermalDetonator, world);
+		thermalDetonatorEntity.setLife(life);
+		thermalDetonatorEntity.setPrimed(primed);
+		thermalDetonatorEntity.onSpawnPacket(new EntitySpawnS2CPacket(thermalDetonatorEntity.getId(), thermalDetonatorEntity.getUuid(), playerEntity.getX(), playerEntity.getY()+1, playerEntity.getZ(), -playerEntity.getPitch(), -playerEntity.getYaw(), thermalDetonatorEntity.getType(),0, new Vec3d(0f, 0f ,0f), playerEntity.getHeadYaw()));
+		return thermalDetonatorEntity;
+	}
+	int ticksToExplosion=1000000;
+	boolean isPrimed=false;
+
+	@Override
+	public void usageTick(World world, LivingEntity user, ItemStack stack, int remainingUseTicks)
+	{
+		((PlayerEntity)user).getItemCooldownManager().getCooldownProgress(this, remainingUseTicks);
+
+		super.usageTick(world, user, stack, remainingUseTicks);
+	}
+
+	@Override
+	public ItemStack finishUsing(ItemStack stack, World world, LivingEntity user)
+	{
+
+		//ThermalDetonatorEntity thermalDetonator = this.createThermalDetonator(world, stack);
+		//thermalDetonator.setLife(1);
+		//thermalDetonator.setPrimed(true);
+		//thermalDetonator.onSpawnPacket(new EntitySpawnS2CPacket(thermalDetonator.getId(), thermalDetonator.getUuid(), user.getX(), user.getY()+1, user.getZ(), -user.getPitch(), -user.getYaw(), thermalDetonator.getType(),0, new Vec3d(0f, 0f ,0f), user.getHeadYaw()));
+		//world.spawnEntity(thermalDetonator);
+		//world.createExplosion(user, user.getX(), user.getY(), user.getZ(), 1f, World.ExplosionSourceType.MOB);
+		PlayerEntity player = (PlayerEntity)user;
+		if(!player.isCreative()){
+			//stack.decrement(1);
+		}
+		return super.finishUsing(stack, world, user);
+	}
+	@Override
+	public void inventoryTick(ItemStack stack, World world, Entity entity, int slot, boolean selected)
+	{
+		if(entity.isPlayer()){
+			PlayerEntity player = (PlayerEntity)entity;
+		}
+		if(this.isPrimed){
+			this.ticksToExplosion--;
+		}
+		if(entity.isInLava()||entity.isOnFire()){
+			PlayerEntity playerEntity = (PlayerEntity)entity;
+			ThermalDetonatorItem ThermalDetonatorItem = (ThermalDetonatorItem)(stack.getItem() instanceof ThermalDetonatorItem ? stack.getItem() : SwgItems.Explosives.ThermalDetonator);
```

Comment:
The first condition should always be true, probably don't need this ternary here, just `(ThermalDetonatorItem)stack.getItem()`